cpp/include/raft/spatial/knn/detail/ann_utils.cuh

```diff
@@ -253,19 +257,19 @@ inline void dots_along_rows(
    */
 }
 
-template <typename T>
+template <typename T, typename LabelT>
 __global__ void accumulate_into_selected_kernel(uint32_t n_rows,
                                                 uint32_t n_cols,
                                                 float* output,
                                                 uint32_t* selection_counters,
                                                 const T* input,
-                                                const uint32_t* row_ids)
+                                                const LabelT* row_ids)
 {
   uint64_t gid = threadIdx.x + (blockDim.x * blockIdx.x);
   uint64_t j   = gid % n_cols;
   uint64_t i   = gid / n_cols;
   if (i >= n_rows) return;
-  uint64_t l = row_ids[i];
+  uint64_t l = (uint64_t)row_ids[i];
```

Comment:
Consider not using C-style conversion (clang-tidy in our config should have generated a warning for this).